### DISCUSSION TITLE
Landing page not working with vite-react-typescript

### DISCUSSION THREAD

**Discussion by we-wake-act21** [eclipse-theia/theia-cloud#183] [Q&A]
While deploying the landing page that was developed using `vite-react-typescript` , so by default vite was configured to generate the `dist` folder with all the `.js` code inside the assets folder. (dist folder structure shown below).

`dist ` 
`      |- assets`
`      |     |- index-adc3sd.js`
`      |     |- app.png`
`      |- index.html`
`      |- logo.svg`

when deploying this landing page to aks cluster, the landing page is unable to access the files within the `assets folder`

the following is the dockerfile used to build the landing page.

`FROM node:16-alpine as build-stage`
`WORKDIR /app`
`COPY package*.json ./ `
`RUN yarn`
`COPY . .`
`RUN yarn build`
`FROM nginx:stable-alpine as production-stage`
`ENV DEBIAN_FRONTEND noninteractive`
`COPY nginx.conf /etc/nginx/conf.d/default.conf`
`COPY --from=build-stage /app/dist /usr/share/nginx/html`
`EXPOSE 80`
`CMD ["nginx", "-g", "daemon off;"] `

How can I resolve this issue?

**Comment by jfaltermeier** [comment]
Hi, unfortunately I am not familiar with Vite. Does a container created with your docker image work on a local machine?
`docker run -p 80:80 your-image` and browser to http://localhost/ ?

**Reply by we-wake-act21** [reply]
yes, the docker image is working on local machine but is not able to access the assets folder when deployed.

**Reply by jfaltermeier** [reply]
Are you using Theia Cloud with paths, e.g. as `https://theia-cloud-url/trynow/` ?
In this case case you might have to tell vite about this `trynow` subdirectory, otherwise it might look for the assets directory at `https://theia-cloud-url/assets/`instead of `https://theia-cloud-url/trynow/assets`: https://stackoverflow.com/questions/75857668/in-vite-how-do-i-specify-a-subdirectory-of-my-server-for-the-assets

**Reply by we-wake-act21** [reply]
I am using the path as `https://theia-cloud-url`

**Reply by jfaltermeier** [reply]
If I understand you correctely your helm values look similar to below?

```
hosts:
  usePaths: true
  paths:
    baseHost: theia-cloud-url
    service: servicex
    landing: ""
    instance: instances
```
    
I think using an empty path segment leads to issues with our landing page ingress, since this will rewrite evertything after the path segment: https://github.com/eclipsesource/theia-cloud-helm/blob/main/charts/theia.cloud/templates/landing-page-ingress-path-based.yaml#L12
Using an empty path segment is not supported at the moment. 

Does your landing page work when the `landing` path segment is not empty?

**Reply by we-wake-act21** [reply]
`service: service.${azurerm_public_ip.host_ip.ip_address}.sslip.io`

  `landing: ${azurerm_public_ip.host_ip.ip_address}.sslip.io`

  `instance: instance.${azurerm_public_ip.host_ip.ip_address}.sslip.io`
  
  this is how I am configuring the ip

**Reply by sgraband** [reply]
Could you double check, whether or not it would work, if you set the landing page to:
`landing: landing.${azurerm_public_ip.host_ip.ip_address}.sslip.io`
?
There might be some problems with an `empty` landing page url in the ingresses, but before we take a look at those we should simply test, whether or not it is really the issue.
